x-pack/plugins/data_enhanced/server/search/session/get_search_session_page.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract, Logger } from 'kibana/server';
+import { from, Observable, EMPTY } from 'rxjs';
+import { concatMap } from 'rxjs/operators';
+import {
+  SearchSessionSavedObjectAttributes,
+  SEARCH_SESSION_TYPE,
+} from '../../../../../../src/plugins/data/common';
+import { CheckSearchSessionsDeps, CheckSearchSessionsFn, SearchSessionsConfig } from './types';
+
+export interface GetSessionsDeps {
+  savedObjectsClient: SavedObjectsClientContract;
+  logger: Logger;
+}
+
+export function getSearchSessionsPage$(
+  { savedObjectsClient }: GetSessionsDeps,
+  filter: any,
+  pageSize: number,
+  page: number
+) {
+  return from(
+    savedObjectsClient.find<SearchSessionSavedObjectAttributes>({
+      page,
+      perPage: pageSize,
+      type: SEARCH_SESSION_TYPE,
+      namespaces: ['*'],
+      // process older sessions first
+      sortField: 'touched',
+      sortOrder: 'asc',
+      filter,
+    })
+  );
+}
+
+export const checkSearchSessionsByPage = (
+  checkFn: CheckSearchSessionsFn,
+  deps: CheckSearchSessionsDeps,
+  config: SearchSessionsConfig,
+  filters: any,
+  nextPage = 1
+): Observable<void> =>
+  checkFn(deps, config, filters, nextPage).pipe(
+    concatMap((result) => {
+      if (!result || !result.saved_objects || result.saved_objects.length < config.pageSize) {
+        return EMPTY;
+      } else {
+        // TODO: while processing previous page session list might have been changed and we might skip a session,
```

Comment:
Maybe new saved object's client API helps with this? https://github.com/elastic/kibana/issues/91175?
Maybe using it would also simplify some of our custom pagination code? (not sure, just worth looking at it)

Reply:
Not a new issue, so I may address in a separate PR